### PR TITLE
fix(renderer): add replaceCurrentUrl method

### PR DESCRIPTION
**Pull request by cbr7**
### What does this PR do?

Adds a new function to be used in a later PR.

First part of fix for https://github.com/podman-desktop/podman-desktop/issues/15994

### How to test?
Only unit using unit tests, otherwise it's not possible the function added has no consumers at this time.

### CODE REVIEW

File: packages/renderer/src/stores/navigation-history.svelte.ts [hunk 1]
```diff
@@ -404,6 +404,25 @@ export function goForward(): void {
...
+ * Navigate to a URL while replacing the current history entry instead of pushing.
+ * Used for auto-redirects (e.g. detail page base path → default tab) so that the
+ * transient redirect URL does not remain in the stack and trap the user in a loop.
+ * @param url - The URL to navigate to
+ * @example
+ * // Stack: ['/', '/containers', '/containers/abc123/'], index 2
+ * replaceCurrentUrl('/containers/abc123/tty')
+ * // Stack: ['/', '/containers', '/containers/abc123/tty'], index 2
+ */
+export function replaceCurrentUrl(url: string): void {
```
Comment (simonrey1) [inline packages/renderer/src/stores/navigation-history.svelte.ts:417]: remark(blocking): I think this should be introduced in a separate PR before being applied to Container and Pod details components.

Comment (cbr7) [reply packages/renderer/src/stores/navigation-history.svelte.ts:417]: That does not make sense, the issue reported is a bug, to fix the bug we need to change the container details page, it makes no sense to split this PR.

Comment (simonrey1) [reply packages/renderer/src/stores/navigation-history.svelte.ts:417]: > the issue reported is a bug, to fix the bug we need to change the container details page

I am not saying the contrary.


> it makes no sense to split this PR.

To me it makes sense cause you are introducing a new function in the navigation history module. Then you use it in Container and Pod details components, but the function can be discussed and approved with domain owners and people who made the module to agree altogether on the new function.

Comment (cbr7) [reply packages/renderer/src/stores/navigation-history.svelte.ts:417]: So you want me to make a PR that does not actually fix the bug that is reported in the original issue, and instead make a PR that contains only half of the fix and then make another PR with the other half? Am I getting that right?

Comment (jiridostal) [reply packages/renderer/src/stores/navigation-history.svelte.ts:417]: @cbr7 Simon's request is to split the PR to an introduction of a standalone function and its first consumer. This is a pattern that we've been following on all other issues and Simon's ask to follow this is valid and with what any other domain owner would do. This is not one-off or anything like that. I'm pretty sure Simon will help you to merge the split PR fast as we are aware that its main purpose is to fix a bug, but conventions should not be skipped because of it. In the end, it's up to the domain owner to give it a green light, and it's not happening with the current state of this PR. 

If you wish to reopen a discussion about PR requirements, feel free to, but not on a single GH ticket.

Comment (cbr7) [reply packages/renderer/src/stores/navigation-history.svelte.ts:417]: @jiridostal ok, so to be very clear, the request is for me to create two different independent PRs, neither of which would by itself fix the bug, and one of which would in effect introduce a new function as dead code, since it wouldn't have any consumers and only after merging the PR with the dead code I should make the PR with the consumers and as such create a dependency between PR 2 and PR 1, is that an accurate summary?

Comment (jiridostal) [reply packages/renderer/src/stores/navigation-history.svelte.ts:417]: Yes that summary is accurate, please go ahead. That is the current process, which will be talked through and possibly adjusted outside of this ticket.

Comment (cbr7) [reply packages/renderer/src/stores/navigation-history.svelte.ts:417]: so be it

Comment (cbr7) [reply packages/renderer/src/stores/navigation-history.svelte.ts:417]: @simonrey1 @jiridostal done, updated the PR to only include the method as dead code.

**Review by simonrey1 — COMMENTED** [review]
Thanks for splitting this out Vladimir! The `replaceCurrentUrl` helper looks clean and the test coverage is solid, nice edge-case handling with the empty stack and extensionEntry stripping. 👍

A couple of small suggestions:

1. Title: Since this PR now only adds the new function (no consumers yet), could we update the title to something like `feat(renderer): add replaceCurrentUrl helper to navigation history store`? The fix() prefix implies the bug is resolved, but the actual fix will come in the follow-up PR.

2. Minor: When the stack is empty, `replaceCurrentUrl` falls through to `router.goto()` without setting `isNavigatingHistory`, so the router subscription ends up pushing the URL instead of replacing. The test documents this, but a short inline comment explaining the intentional fallthrough would help future readers. Alternatively, we could `return` early if the stack is empty and let the caller handle that case explicitly.

Looks good overall!

File: packages/renderer/src/stores/navigation-history.svelte.ts [hunk 2]
```diff
@@ -404,6 +404,25 @@ export function goForward(): void {
...
+  if (navigationHistory.index >= 0 && navigationHistory.index < navigationHistory.stack.length) {
+    navigationHistory.stack[navigationHistory.index] = { url };
+    navigationHistory.stack = [...navigationHistory.stack];
```
Comment (benoitf) [inline packages/renderer/src/stores/navigation-history.svelte.ts]: out of curiosity, do we need the spread here as navigationHistory is a state


side question: do we need safeguard around "url" being the current url ?

Comment (cbr7) [reply packages/renderer/src/stores/navigation-history.svelte.ts]: This is how it's currently used in several other places in the file, so I would say yes.

For the second question, I guess the answer is technically a safeguard isn't strictly needed, but I've added one just to be on the safe side.

Comment (benoitf) [reply packages/renderer/src/stores/navigation-history.svelte.ts]: it looks in other references it's to add new elements
but here it seems you can do
```
navigationHistory.stack.map((entry, i) =>
  i === navigationHistory.index ? { url } : entry
);
```

Comment (cbr7) [reply packages/renderer/src/stores/navigation-history.svelte.ts]: fair, updated.

**Review by benoitf — APPROVED** [review]

**Comment by benoitf** [comment]
waiting for @simonrey1 feedback before merging I would say

**Review by simonrey1 — APPROVED** [review]
